File: algorithm/BinarySearchTree/BST.js
	class Node {
	
constructor(item) {
			this.key = item;
			this.left = this.right = null;
		}
	}

	var root = null;

	function insert(key) {
		root = insertRec(root, key);
	}

	function insertRec(root, key) {

		if (root == null) {
			root = new Node(key);
			return root;
		}

		if (key < root.key)
			root.left = insertRec(root.left, key);
		else if (key > root.key)
			root.right = insertRec(root.right, key);

		return root;
	}

	function inorder() {
		inorderRec(root);
	}


	function inorderRec(root)
	{
		if (root != null) {
            console.log(root.key);
			inorderRec(root.left);
			inorderRec(root.right);
		}
	}


		insert(50);
		insert(30);
		insert(20);
		insert(40);
		insert(70);
		insert(60);
		insert(80);


		inorder();

File: algorithm/BinarySearchTree/BST.java
public class BST {
    public Node root;
    public BST(int x){
        this.root=addnode(root,x);
    }
    public BST(){
        this.root=null;
    }
    public Node addnode(Node root,int x){
        if (root== null ){
            root = new Node(x);
        }
        else if(root.data>=x){
            root.right=addnode(root.right,x);
        }
        else{
            root.left=addnode(root.left,x);
        }
        return root;
    }

    public void add(int x){
        addnode(this.root,x);
    }

    public Node find(Node root,int x){
        if (root== null ){
            return null;
        }
        if (root.data== x ){
            return root;
        }
        else if(root.data>=x){
            return  find(root.right,x);
        }
        else{
            return  find(root.left,x);
        }
    }

    public Node findit(int x){
        return find(this.root,x);
    }

    
}
